fixes/consolidate-multiple-richtext-history.sql

```diff
@@ -0,0 +1,157 @@
+/*
+ *	Consolidate field history records, on configurable time threshold (@timeThreshold) in minutes.
+ *
+ *	NOTE:  This script defaults to rolling back changes.
+ *		To commit changes, set @saveChanges = 1.
+ */
+declare @saveChanges bit; --set @saveChanges = 1
+declare @TableName varchar(100); --set @TableName = 'Test'; --Table Name Ex: Test, RegressionTest, BaseAsset
+declare @Table varchar(100); set @Table = quotename(@TableName);
+declare @Table_Now varchar(100); set @Table_Now = quotename(@TableName + '_Now');
+declare @FieldName varchar(100); --set @FieldName = 'ExpectedResults'; -- Table attribute Ex: Description, ExpectedResults, Setup
+declare @Field varchar(100); set @Field = quotename(@FieldName);
+declare @timeThreshold int;
+
+set @timeThreshold = 5
+
+create table #assets(ID int not null, CurrentAuditID int not null, NextAuditID int not null)
+create table #suspect(ID int not null, CurrentAuditID int not null, NextAuditID int not null)
+create table #bad(ID int not null, CurrentAuditID int not null, NextAuditID int not null)
+
+set nocount on; begin tran; save tran TX
+
+-- Asset Rows
+declare @q1 varchar(max);
+select @q1 = ';
+with H as (
+	select AA.ID, AA.AssetType, AuditID, R=ROW_NUMBER() OVER(partition by AA.ID, AA.AssetType order by AuditID)
+	from dbo.AssetAudit AA
+)
+insert #assets(ID, CurrentAuditID, NextAuditID)
+select aH.ID, aH.AuditID, bH.AuditID
+from H as aH
+join dbo.' + @Table + ' aAsset on aAsset.ID=aH.ID and aAsset.AssetType=aH.AssetType and aH.AuditID = aAsset.AuditBegin
+join H as bH on bH.ID=aH.ID and aH.AssetType=bH.AssetType and bH.R=aH.R+1
+join dbo.' + @Table + ' bAsset on bAsset.ID=bH.ID and bAsset.AssetType=bH.AssetType and bH.AuditID = bAsset.AuditBegin'
+
+-- Consecutive Asset changes
+declare @q2 varchar(max);
+select @q2 = ';
+insert #suspect(ID, CurrentAuditID, NextAuditID)
+select Assets.ID, Assets.CurrentAuditID, Assets.NextAuditID
+from #assets as Assets
+join dbo.' + @Table + ' currentAsset on currentAsset.ID=Assets.ID and currentAsset.AuditBegin=Assets.CurrentAuditID
+join dbo.' + @Table + ' nextAsset on nextAsset.ID=Assets.ID and nextAsset.AuditBegin=Assets.NextAuditID
+join dbo.Audit currentA on currentA.ID = Assets.CurrentAuditID
+join dbo.Audit nextA on nextA.ID = Assets.NextAuditID
+WHERE
+ISNULL(currentA.ChangedByID,-1) = ISNULL(nextA.ChangedByID,-1)  -- Consecutive changes from the same user
+AND DATEDIFF(mi,currentA.ChangeDateUTC,nextA.ChangeDateUTC) <= ' + CAST(@timethreshold as varchar(10)) + ' --Time threshold / period / lapse.
```

Comment:
Be aware that `DATEDIFF()`, despite its name, _does not calculate the actual difference_ between 2 times.  This check could be true for differences almost a minute longer than the threshold, depending on _where in the minute_ the times fall.

Maybe that's acceptable?  For tighter tolerance, use `DATEDIFF_BIG(ms, ...`

Reply:
Actually, I think you just want `<`, not `<=`.  Simple

Reply:
It probably doesn't matter...

Reply:
It's the difference between `5` meaning "less than 6" or meaning "less than 5" 🤷🏼

Reply:
No, I mean in the grand scheme of things, it probably doesn't matter - we're trying to drop "noise" changes, and tweak the time window to create desired blast radius given actual shape of customer's data.

Reply:
No problem, as long as the tweaker understands that "5" is "5:59.997".